### PR TITLE
Added support for empty context in delete API

### PR DESCRIPTION
**Pull request by singhalprerana**
Added DEFAULT_CONTEXT as default when context is empty to delete corresponding config

### CODE REVIEW

**Review by aaron-steinfeld — COMMENTED** [review]
Since it's failing on the build and only a one line fix, do you mind extending the expiration on the snyk ignore in `.snyk` ?

LGTM - @GurtejSohi just to confirm, there was no reason this was missing other than we didn't expect to want to do this, right?

**Comment by singhalprerana** [comment]
> Since it's failing on the build and only a one line fix, do you mind extending the expiration on the snyk ignore in `.snyk` ?
Done.. but what is this exactly? if we are anyways ignoring, why add it in first place?

**Comment by aaron-steinfeld** [comment]
> > Since it's failing on the build and only a one line fix, do you mind extending the expiration on the snyk ignore in `.snyk` ?
> 
> Done.. but what is this exactly? if we are anyways ignoring, why add it in first place?

So our only two options are to ignore it or fix it. So we don't permanently ignore it because it is a vulnerability we'd like to fix - https://snyk.io/vuln/SNYK-JAVA-IONETTY-1042268 , but can't do so yet as there's no fix available for the upstream (sounds from their github that it won't be until the next major release). The expiration is kind of a reminder for us to periodically check to see if there is a fix available - we usually push it out 3ish months at a time.

File: .snyk [hunk 1]
```diff
@@ -5,5 +5,5 @@ ignore:
   SNYK-JAVA-IONETTY-1042268:
     - '*':
         reason: No replacement available
-        expires: 2021-08-31T00:00:00.000Z
+        expires: 2021-09-02T00:00:00.000Z
```
Comment (aaron-steinfeld) [inline .snyk:8]:
```suggestion
        expires: 2021-12-31T00:00:00.000Z
```

**Comment by jcchavezs** [comment]
@singhalprerana Just curious about why this is needed, in which cases we have such context?

**Comment by singhalprerana** [comment]
> @singhalprerana Just curious about why this is needed, in which cases we have such context?

We store configs w.r.t. customers - we also want to delete them given customer use-cases like reset to default (e.g. new risk scoring we are currently working on)... Current API allows the config to gets stored wrt default_context. But there is no way to delete it..

**Comment by GurtejSohi** [comment]
> LGTM - @GurtejSohi just to confirm, there was no reason this was missing other than we didn't expect to want to do this, right?

That's right. With this change, we should also update [this documentation comment](https://github.com/hypertrace/config-service/blob/main/config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto#L102) - now the context is optional. @singhalprerana can you please update this?

**Comment by aaron-steinfeld** [comment]
> > @singhalprerana Just curious about why this is needed, in which cases we have such context?
> 
> We store configs w.r.t. customers - we also want to delete them given customer use-cases like reset to default (e.g. new risk scoring we are currently working on)... Current API allows the config to gets stored wrt default_context. But there is no way to delete it..

Slightly more broadly - some config has its own ID (imagine something like user preferences), other config could be global to the tenant (such as something like agent configuration). In the config model, that global scenario means it has a resource name/namespace, but no specific context - it's global. As @singhalprerana mentioned, we can write to and read that tenant-wide config, but we can't remove it. I can't recall if that's intentional, assuming that we'd never want to have no config, or just a bug, but even if it was intentional, it doesn't match how we often use it for this type of scenario. That is, loading a default value from config, and only storing an override - so a "reset to default" use case would involve deleting the override.

**Comment by singhalprerana** [comment]
> > LGTM - @GurtejSohi just to confirm, there was no reason this was missing other than we didn't expect to want to do this, right?
> 
> That's right. With this change, we should also update [this documentation comment](https://github.com/hypertrace/config-service/blob/main/config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto#L102) - now the context is optional. @singhalprerana can you please update this?

done

**Review by aaron-steinfeld — APPROVED** [review]

**Review by GurtejSohi — APPROVED** [review]

**Pull request merged**